packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -552,6 +555,31 @@ mixin WidgetsBinding on BindingBase, ServicesBinding, SchedulerBinding, GestureB
         },
       );
 
+      registerServiceExtension(name: 'invokePreHotRestartCallbacks', callback: (Map<String, Object> params) async {
+        Future<void> invokeAndWait(DebugPreHotRestartCallback callback, String label) async {
+          developer.postEvent('preHotRestartCallback', <String, Object>{'label': label, 'finished': false});
+          try {
+            await Future<Object?>.value(callback());
+          } catch (error, stack) {
+            FlutterError.reportError(
+              FlutterErrorDetails(
+                exception: error,
+                stack: stack,
+                context: ErrorSummary('Failed to invoke preHotRestartCallback "$label"'),
+              )
+            );
+          } finally {
+            developer.postEvent('preHotRestartCallback', <String, Object>{'label': label, 'finished': true});
```

Comment:
what is the purpose of this event? I couldn't find anything on the tool side that listens to it, did I miss something?